api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/MetricsServiceLifecycle.java

```diff
@@ -379,10 +384,16 @@ private int getDefaultTTL() {
     }
 
     private void initJobsService() {
+        RxSession rxSession = new RxSessionImpl(session);
         jobsService = new JobsServiceImpl();
         jobsService.setMetricsService(metricsService);
-        jobsService.setSession(new RxSessionImpl(session));
-        jobsService.setScheduler(new SchedulerImpl(new RxSessionImpl(session)));
+        jobsService.setSession(rxSession);
+        if (schedulerMode.equals("test")) {
```

Comment:
To overlay the class means I would have to maintain a separate copy of it in the metrics-api-jaxrs-test module. Having to keep the two source files in sync seems too error prone. I have added a layer of indirection for creating the scheduler instance with a new class JobSchedulerFactory, which does get replaced in the metrics-api-jaxrs-test module.

The config/system property that I initially used to activate using TestScheduler has been removed, which I think is the most important thing.